zipkin/src/main/java/zipkin2/storage/StorageComponent.java

```diff
@@ -68,6 +70,33 @@ public ServiceAndSpanNames serviceAndSpanNames() { // delegates to deprecated me
 
   public abstract SpanConsumer spanConsumer();
 
+  /**
+   * A storage request failed and was dropped due to a limit, resource unavailability, or a timeout.
+   * Implementations of throttling can use this signal to differentiate between failures, for
+   * example to reduce traffic.
+   *
+   * <p>Callers of this method will submit an exception raised by {@link Call#execute()} or on the
+   * error callback of {@link Call#enqueue(Callback)}.
+   *
+   * <p>By default, this returns true if the input is a {@link RejectedExecutionException}. When
+   * originating exceptions, use this type to indicate a load related failure.
+   *
+   * <p>Given a 3rd party exception, re-wrapping to force {@link RejectedExecutionException} seems
```

Comment:
I was a bit confused what automatic handling meant here since this code itself is a bit far from the actual handling. Maybe can be more opinionated:

```
It is generally preferred to specialize this method to handle relevant exceptions for the particular storage rather than wrapping them in `RejectedExecutionException` at call sites. Extra wrapping can make errors harder to read, for example, by making it harder to "google" a solution for a well known error message for the storage client, instead thinking the error is in Zipkin code itself.
```

Reply:
agree this is is better text!